test/cpp/util/error_details_test.cc

```diff
@@ -98,7 +98,7 @@ TEST(SetTest, OutOfScopeErrorCode) {
 }
 
 TEST(SetTest, ValidScopeErrorCode) {
-  for (int c = StatusCode::OK; c <= StatusCode::UNAUTHENTICATED; c++) {
+  for (int c = StatusCode::CANCELLED; c <= StatusCode::UNAUTHENTICATED; c++) {
```

Comment:
I'm not sure why do we need to align this test with the HTTP2 specification. Is this causing issue with moving grpc::Status to absl::Status?